librz/analysis/arch/arm/arm_il64.c

```diff
@@ -0,0 +1,2713 @@
+ */
+static RzILOpEffect *write_reg(arm64_reg reg, RZ_OWN RZ_NONNULL RzILOpBitVector *v) {
+	rz_return_val_if_fail(v, NULL);
+	const char *var = reg_var_name(reg);
+	if (!var) {
+		rz_il_op_pure_free(v);
+		return NULL;
+	}
+	if (is_wreg(reg)) {
+		v = UNSIGNED(64, v);
+	}
+	return SETG(var, v);
+}
+
+static RzILOpBitVector *arg_mem(RzILOpBitVector *base_plus_disp, cs_arm64_op *op) {
+	if (op->mem.index == ARM64_REG_INVALID) {
+		return base_plus_disp;
+	}
+	RzILOpBitVector *index = read_reg(op->mem.index);
+	index = extend(64, op->ext, index, reg_bits(op->mem.index));
+	index = apply_shift(op->shift.type, op->shift.value, index);
+	return ADD(base_plus_disp, index);
+}
+
+/**
+ * IL to retrieve the value of the \p n -th arg of \p insn
+ * \p bits_inout Setting the backing variable to non-0 indicates that the result must have this bitness.
+ *               This is necessary for immediate operands for example.
+ *               In any case, if a value is returned, its bitness is written back into this storage.
+ */
+static RzILOpBitVector *arg(cs_insn *insn, size_t n, ut32 *bits_inout) {
+	ut32 bits_requested = bits_inout ? *bits_inout : 0;
+	cs_arm64_op *op = &insn->detail->arm64.operands[n];
+	switch (op->type) {
+	case ARM64_OP_REG: {
+		if (!bits_requested) {
+			bits_requested = REGBITS(n);
+			if (bits_inout) {
+				*bits_inout = bits_requested;
+			}
+		}
+		RzILOpBitVector *r = REG(n);
+		if (!r) {
+			return NULL;
+		}
+		return apply_shift(op->shift.type, op->shift.value, extend(bits_requested, op->ext, r, REGBITS(n)));
+	}
+	case ARM64_OP_IMM: {
+		if (!bits_requested) {
+			return NULL;
+		}
+		ut64 val = IMM(n);
+		if (op->shift.type == ARM64_SFT_LSL) {
+			val <<= op->shift.value;
+		}
+		return UN(bits_requested, val);
+	}
+	case ARM64_OP_MEM: {
+		RzILOpBitVector *addr = MEMBASE(n);
+		st64 disp = MEMDISP(n);
+		if (disp > 0) {
+			addr = ADD(addr, U64(disp));
+		} else if (disp < 0) {
+			addr = SUB(addr, U64(-disp));
+		}
+		return arg_mem(addr, &insn->detail->arm64.operands[n]);
+	}
+	default:
+		break;
+	}
+	return NULL;
+}
+
+#define ARG(n, bits) arg(insn, n, bits)
+
+/**
+ * zf := v == 0
+ * nf := msb v
+ */
+static RzILOpEffect *update_flags_zn(RzILOpBitVector *v) {
+	return SEQ2(
+		SETG("zf", IS_ZERO(v)),
+		SETG("nf", MSB(DUP(v))));
+}
+
+/**
+ * zf := v == 0
+ * nf := msb v
+ * cf := 0
+ * vf := 0
+ */
+static RzILOpEffect *update_flags_zn00(RzILOpBitVector *v) {
+	return SEQ3(
+		update_flags_zn(v),
+		SETG("cf", IL_FALSE),
+		SETG("vf", IL_FALSE));
+}
+
+/**
+ * Capstone: ARM64_INS_ADD, ARM64_INS_ADC, ARM64_INS_SUB, ARM64_INS_SBC
+ * ARM: add, adds, adc, adcs, sub, subs, sbc, sbcs
+ */
+static RzILOpEffect *add_sub(cs_insn *insn) {
+	if (!ISREG(0)) {
+		return NULL;
+	}
+#if 0
+	if ((insn->id == ARM64_INS_ADD || insn->id == ARM64_INS_SUB) &&
+		!insn->detail->arm.update_flags && OPCOUNT() == 3 && REGID(1) == ARM64_REG_PC && ISIMM(2)) {
+		// alias for adr
+		return adr(insn, is_thumb);
+	}
+#endif
```

Comment:
why this is removed?

Reply:
I don't remember, and since there is no comment with more info, no use in keeping in my opinion.